contracts/Strategy.sol

```diff
@@ -256,63 +347,45 @@ contract Strategy is BaseStrategy {
-        if (isUniswap) {
-            router = uniswapRouter;
-        } else {
-            router = sushiswapRouter;
-        }
-    }
-
     // Allow external debt repayment
-    // Attempt to take currentRatio to target c-ratio
-    // Passing zero will repay all debt if possible
-    function emergencyDebtRepayment(uint256 currentRatio)
+    // Allow repayment of an arbitrary amount of MIM in case of an emergency
+    // If `amount` is larger than 0, the function will attempt to just repay using MIM
+    // disregarding straty logic.  This could be helpful if for example yvMIM withdrawals are failing and
+    // we want to do a MIM airdrop and direct debt repayment instead
+    // If `amount` is 0, it will attempt to take currentRatio to target c-ratio
+    // Passing zero in `currentRatio` will repay all debt if possible.
+    function emergencyDebtRepayment(uint256 amount, uint256 currentRatio)
         external
         onlyVaultManagers
     {
-        _repayDebt(currentRatio);
+        if (amount > 0) {
+            _repayInvestmentTokenDebt(amount);
+        } else {
+            _repayDebt(currentRatio);
+        }
     }
 
-    // Allow repayment of an arbitrary amount of Dai without having to
-    // grant access to the CDP in case of an emergency
-    // Difference with `emergencyDebtRepayment` function above is that here we
-    // are short-circuiting all strategy logic and repaying Dai at once
-    // This could be helpful if for example yvDAI withdrawals are failing and
-    // we want to do a Dai airdrop and direct debt repayment instead
-    function repayDebtWithDaiBalance(uint256 amount)
-        external
-        onlyVaultManagers
-    {
-        _repayInvestmentTokenDebt(amount);
-    }
 
-    // ******** OVERRIDEN METHODS FROM BASE CONTRACT ************
+    // ******** OVERRIDDEN METHODS FROM BASE CONTRACT ************
 
     function name() external view override returns (string memory) {
         return strategyName;
     }
 
     function delegatedAssets() external view override returns (uint256) {
-        return _convertInvestmentTokenToWant(_valueOfInvestment());
+        return _convertInvestmentTokenToWant(_valueOfInvestment().add(balanceOfInvestmentTokenInBentoBox()).add(
+            balanceOfInvestmentToken()));
     }
 
     function estimatedTotalAssets() public view override returns (uint256) {
-        return
-            balanceOfWant()
-                .add(balanceOfMakerVault())
-                .add(_convertInvestmentTokenToWant(balanceOfInvestmentToken()))
-                .add(_convertInvestmentTokenToWant(_valueOfInvestment()))
-                .sub(_convertInvestmentTokenToWant(balanceOfDebt()));
+        uint256 remainingInvestmentToken =
+            _valueOfInvestment()
+                .add(balanceOfInvestmentTokenInBentoBox())
+                .add(balanceOfInvestmentToken())
+                .add(balanceOfCollateralInMIM())
```

Comment:
When you do abracadabra.addCollateral, it will take your collateral/want from bentobox and transfer it to them, so you don't have it anymore on your bentobox balance. ``balanceOfWantInBentoBox()`` is the balance of want that has not been added as a collateral.